pinot-core/src/main/java/org/apache/pinot/core/query/selection/SelectionOperatorUtils.java

```diff
@@ -647,4 +649,42 @@ public static <T> void addToPriorityQueue(T value, PriorityQueue<T> queue, int m
       queue.offer(value);
     }
   }
+
+  public static <T> T arrangeColumnsToMatchProjection(DataSchema dataSchema, Iterator<Object[]> rows,
```

Comment:
Suggest not extracting the whole method because it is not as readable, and not very reusable. It is like forcefully stitching 2 methods together, with potential performance overhead of using iterator, passing function and extra per-record if check.
We can extract the part of creating `resultDataSchema`: `public static DataSchema arrangeColumns(DataSchema dataSchema, int[] columnIndices)`